File: designtime/webstudio/ui/src/app/editables/rule-template-instance/differ/MergedDiffModificationEntry.ts
import { ModificationEntry } from './ModificationEntry';
import { ModificationType } from './ModificationType';

export class MergedDiffModificationEntry extends ModificationEntry {

  private _message: string;

  private _localVersion: string;
  private _baseVersion: string;

  private _localVersionObj: Object;
  private _baseVersionObj: Object;

  private _localChangeType: ModificationType;
  private _serverChangeType: ModificationType;

  constructor(modificationType: ModificationType, localChangeType: ModificationType, serverChangeType: ModificationType) {
    super(modificationType);
    this.localChangeType = localChangeType;
    this.serverChangeType = serverChangeType;
  }

  get message(): string {
    return this._message;
  }

  set message(value: string) {
    this._message = value;
  }

  get localVersion(): string {
    return this._localVersion;
  }

  set localVersion(value: string) {
    this._localVersion = value;
  }

  get baseVersion(): string {
    return this._baseVersion;
  }

  set baseVersion(value: string) {
    this._baseVersion = value;
  }

  get localVersionObj(): Object {
    return this._localVersionObj;
  }

  set localVersionObj(value: Object) {
    this._localVersionObj = value;
  }

  get baseVersionObj(): Object {
    return this._baseVersionObj;
  }

  set baseVersionObj(value: Object) {
    this._baseVersionObj = value;
  }

  get localChangeType(): ModificationType {
    return this._localChangeType;
  }

  set localChangeType(value: ModificationType) {
    this._localChangeType = value;
  }

  get serverChangeType(): ModificationType {
    return this._serverChangeType;
  }

  set serverChangeType(value: ModificationType) {
    this._serverChangeType = value;
  }

  public isLocalChange(): boolean {
    return this.localChangeType != null || this.localChangeType === ModificationType.UNCHANGED;
  }

  public isServerChange(): boolean {
    return this.serverChangeType != null || this.serverChangeType === ModificationType.UNCHANGED;
  }

}

File: designtime/webstudio/WebStudio/src/com/tibco/cep/webstudio/client/diff/MergedDiffModificationEntry.java
package com.tibco.cep.webstudio.client.diff;

/**
 * Class to represent a difference while showing DIFF between 3 versions (Base,
 * User, Server) of an artifact.
 * 
 * @author moshaikh
 * 
 */
public class MergedDiffModificationEntry extends ModificationEntry {

	/**
	 * Message explaining the nature of diff (eg: 'Modified locally and on Server', 'Added locally'..)
	 */
	private String message;
	private String localVersion;
	private String baseVersion;
	
	private Object localVersionObj;
	private Object baseVersionObj;
	
	private ModificationType localChangeType;
	private ModificationType serverChangeType;

	/**
	 * 
	 * @param modificationType The change type to be depicted to user. (The CSS color depends on this.)
	 * @param localChangeType The type of change done on local.
	 * @param serverChangeType The type of change done on server side.
	 */
	public MergedDiffModificationEntry(ModificationType modificationType, ModificationType localChangeType, ModificationType serverChangeType) {
		super(modificationType);
		this.localChangeType = localChangeType;
		this.serverChangeType = serverChangeType;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}

	public String getLocalVersion() {
		return localVersion;
	}

	public void setLocalVersion(String localVersion) {
		this.localVersion = localVersion;
	}

	public String getBaseVersion() {
		return baseVersion;
	}

	public void setBaseVersion(String baseVersion) {
		this.baseVersion = baseVersion;
	}

	public Object getLocalVersionObj() {
		return localVersionObj;
	}

	public void setLocalVersionObj(Object localVersionObj) {
		this.localVersionObj = localVersionObj;
	}

	public Object getBaseVersionObj() {
		return baseVersionObj;
	}

	public void setBaseVersionObj(Object baseVersionObj) {
		this.baseVersionObj = baseVersionObj;
	}

	public boolean isLocalChange() {
		return localChangeType != null || localChangeType == ModificationType.UNCHANGED;
	}

	public boolean isServerChange() {
		return serverChangeType != null || serverChangeType == ModificationType.UNCHANGED;
	}

	public ModificationType getLocalChangeType() {
		return localChangeType;
	}

	public void setLocalChangeType(ModificationType localChangeType) {
		this.localChangeType = localChangeType;
	}

	public ModificationType getServerChangeType() {
		return serverChangeType;
	}

	public void setServerChangeType(ModificationType serverChangeType) {
		this.serverChangeType = serverChangeType;
	}
}
